plugins/http_plugin/include/eosio/http_plugin/beast_http_listener.hpp

```diff
@@ -107,7 +109,13 @@ class beast_http_listener : public std::enable_shared_from_this<beast_http_liste
       auto self = this->shared_from_this();
       acceptor_.async_accept(socket_, [self](beast::error_code ec) {
          if(ec) {
-            fail(ec, "accept", self->plugin_state_->logger, "closing connection");
+            if (ec == boost::system::errc::too_many_files_open) {
+               fail(ec, "accept", self->plugin_state_->logger, "too many files open - waiting 500ms");
+               self->timer_.expires_from_now(boost::posix_time::milliseconds(500));
```

Comment:
`boost::asio::deadline_timer` is not thread safe. The `async_accept` handler is called from the thread pool executor.

Reply:
I wonder if a simple escalating sleep amount would be fine here with a max of say 500ms.

Reply:
> `boost::asio::deadline_timer` is not thread safe. The `async_accept` handler is called from the thread pool executor.

I was thinking, instead of `boost::asio::deadline_timer`, maybe I could just use `std::this_thread::sleep_for(std::chrono::milliseconds(500));`

About the thread safety remark, I think we have only one listener object (for each type) waiting on accept(), right?

Reply:
> I wonder if a simple escalating sleep amount would be fine here with a max of say 500ms.

I don't see a good benefit if adding complexity with an escalating delay. I think because we reach a state where all file handles are in use, it means that we are accepting connections faster than we are able to process the requests from these connections. So pausing the accept for 500ms to give nodeos a chance to clear the backlog a bit doesn't seem excessive to me, and valid flow control. But maybe I'm missing a subtelty here?

Reply:
Ah, I think you are correct about the safety remark. Never mind.

Reply:
Use std::chrono instead of boost